src/codec.rs

```diff
@@ -601,6 +601,90 @@ macro_rules! impl_for_non_zero {
 		)*
 	}
 }
+
+/// Encode the slice without prepending the len.
+///
+/// This is equivalent to encoding all the element one by one, but it is optimized for some types.
+pub(crate) fn encode_slice_no_len<T: Encode, W: Output>(slice: &[T], dest: &mut W) {
```

Comment:
the code of this function is just copy pasted from encode of arrays which must be same as encode of `&[T]`.